Automattic/wp-calypso
# Emails: Switching site does not update domain name in emails home

#### Changes proposed in this Pull Request

This change request is connecting two components to react state so when updating siteID, we update also relative data, like domains.

#### Testing instructions

1. Have two sites with no email plans (but domains)
2. Go to Upgrades -> Emails
3. Notice the Email address in the form for Professional Email and Google Workspace.
4. Change to the other site with "< Switch Site" button in the left top corner.
5. Notice the Email address in the previous forms has updated.

#### REMARKS
The related Asana ticket, is only referencing the problem with Professional Email, but I realised that it also happens with Google Workspace form, so I've updated the code in both cases.

Related to {1200182182542585-as-1201117888572949}
